Add MultiversX namespace (#88)
ChainAgnostic/namespaces: mvx/caip2.md

---
namespace-identifier: mvx-caip2
title: MultiversX Namespace - Chains
author: MultiversX Team <contact@multiversx.com>
discussions-to: https://github.com/ChainAgnostic/namespaces/issues/89
status: Draft
type: Standard
created: 2023-09-11
requires: ["CAIP-2"]
---

# CAIP-2

_For context, see the [CAIP-2][] specification._

## Abstract

In CAIP-2 a general blockchain identification scheme is defined. This is the

> implementation of CAIP-2 for the MultiversX network.

Reviewer: ```suggestion
profile of CAIP-2 specific to the MultiversX network.
```

### MultiversX Namespace

The namespace `mvx` refers to the wider MultiversX ecosystem.

## Rationale

MultiversX consists of multiple networks: a production network (Mainnet), a development testing network (Devnet)
and a testing network (Testnet).

## Syntax

An identifier for an MultiversX chain consists of the namespace prefix "mvx:"

> followed by the chain's Id.

Reviewer: ```suggestion
followed by the appropriate chainId corresponding to that chain.
```

### Reference Definition

> The reference on the MultiversX chain is the equivalent of the MultiversX chainId:

Reviewer: ```suggestion
The reference for each MultiversX chain is identical to its native representation in the MultiversX ecosystem:
```

`1` for `Mainnet`, `D` for `Devnet`, `T` for `Testnet`

> 

Reviewer: How are these values governed? Is there some kind of governance around future chainIds for future chains?  Will they always be single ASCII characters? Can someone spin up a private network and use some other character, and if so, would it be conformant to use, say, `X` if the dapp can learn (out of bound) how and where to connect to an X-network node?

Author: No, the chainID value is not governed for the time being. Since there is only one mainnet chain (for now), the chainIDs for the test chains (testnet, devnet, etc) were decided socially so that the community could go with them.
More about the currently available constants can be found here: [https://docs.multiversx.com/developers/constants](https://docs.multiversx.com/developers/constants)
Once the current convention changes through a MIP(MultiversX Improvement Proposal) and a new standard is settled upon, this section will be updated.
